tests/cpp/test_host_ir_integration.cpp

```diff
@@ -287,6 +287,41 @@ TEST_F(HostIrIntegrationTest, InsertDeallocations) {
       << ") was higher than expected << (" << kExpectedPeakMemory << ")";
 }
 
+TEST_F(HostIrIntegrationTest, ExcludeOutputsFromDeallocations) {
+  c10::DeviceIndex device_index = 0;
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  std::vector<int64_t> input_shape{8, 8};
+  auto in = TensorViewBuilder()
+                .ndims(input_shape.size())
+                .dtype(DataType::Double)
+                .build();
+
+  fusion->addInput(in);
+  TensorView* t0 = add(in, in);
+  TensorView* t1 = matmul(t0, t0);
+  fusion->addOutput(t0);
+  fusion->addOutput(t1);
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+
+  at::Tensor in_tensor = at::randn(
+      input_shape, at::dtype(at::kDouble).device(at::kCUDA, device_index));
+  auto out_tensors = executor_cache.runFusionWithInputs({in_tensor});
+
+  FusionKernelRuntime* runtime = executor_cache.getMostRecentKernelRuntime();
+  EXPECT_EQ(runtime->getHostIrEvaluator().canRun(), "");
```

Comment:
```suggestion
```
I'll create another PR making canRun part of `run` or the constructor. It's silly to ask users to remember to call canRun before run.